File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:

        queue = deque()

        level = 0
        if root is None:
            return []
        queue.append(root)
        ans = [[root.val]]
        while len(queue) > 0:
            tmp = []
            for i in range(len(queue)):
                node = queue.popleft()

                if node.left:
                    queue.append(node.left)
                    tmp.append(node.left.val)
                if node.right:
                    queue.append(node.right)
                    tmp.append(node.right.val)
            if len(tmp)>0:
                ans.append(tmp)
                level += 1

        return ans

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:

    void dfs(TreeNode* root,int height, vector<int> &ans){
        if(root==nullptr){
            return ;
        }
        if(height==ans.size()){
            ans.push_back(root->val);
        }
        dfs(root->right, height+1,ans);
        
        dfs(root->left, height+1, ans);
        return;
    }

    vector<int> rightSideView(TreeNode* root) {

        vector<int> ans={};

        dfs(root, 0 ,ans);
        //reverse(ans.begin(), ans.end());
        return ans;
        
    }
};
